common/src/main/java/software/bernie/geckolib/animation/EasingType.java

```diff
@@ -377,4 +380,48 @@ static Double2DoubleFunction step(Double n) {
 			return leftBorderIndex * stepLength;
 		};
 	}
+
+	class CatmullRomEasing implements EasingType {
+		/**
+		 * Performs a Catmull-Rom interpolation, used to get smooth interpolated motion between keyframes {@link Mth#catmullrom}
+		 */
+		static double catmullRom(double time, double controlPoint1, double controlPoint2, double controlPoint3, double controlPoint4) {
+			return 0.5 * (2.0 * controlPoint2 + (controlPoint3 - controlPoint1) * time +
+					(2.0 * controlPoint1 - 5.0 * controlPoint2 + 4.0 * controlPoint3 - controlPoint4) * time * time +
+					(3.0 * controlPoint2 - controlPoint1 - 3.0 * controlPoint3 + controlPoint4) * time * time * time);
+		}
+
+		@Override
+		public Double2DoubleFunction buildTransformer(Double value) {
+			return easeInOut(EasingType::catmullRom);
+		}
+
+		public double apply(AnimationPoint animationPoint, Double easingValue, double lerpValue) {
+			if (animationPoint.currentTick() >= animationPoint.transitionLength())
+				return (float)animationPoint.animationEndValue();
+
+			List<? extends MathValue> easingArgs = animationPoint.keyFrame().easingArgs();
+
+			if (easingArgs.size() < 2)
+				return Mth.lerp(buildTransformer(easingValue).apply(lerpValue), animationPoint.animationStartValue(), animationPoint.animationEndValue());
+
+			return catmullRom(lerpValue, easingArgs.get(0).get(), animationPoint.animationStartValue(), animationPoint.animationEndValue(), easingArgs.get(1).get());
+		}
+	}
+
+	class SingleStepEasing implements EasingType {
+		@Override
+		public Double2DoubleFunction buildTransformer(Double value) {
+			return time -> time == 1 ? 1 : 0;
+		}
+
+		public double apply(AnimationPoint animationPoint, Double easingValue, double lerpValue) {
+			if (animationPoint.currentTick() >= animationPoint.transitionLength())
+				return (float)animationPoint.animationEndValue();
+
+			if (easingValue == null) easingValue = animationPoint.animationStartValue();
+
+			return lerpValue == 1 ? animationPoint.animationEndValue() : easingValue;
```

Comment:
Why does the SingleStep easing use the input easingvalue as its default value? This would require your input value in the json to be _exactly_ what you want, and would vary between rotation, scale, translation, etc..

Why is it this way specifically?